### PR TITLE
fix: cookie decoding for Node and Cloudflare

### PR DESCRIPTION
**Pull request by vicb**
fixes https://github.com/opennextjs/opennextjs-aws/issues/888

### CODE REVIEW

File: packages/open-next/src/http/util.ts [hunk 1]
```diff
@@ -28,7 +28,13 @@ export const convertHeader = (header: http.OutgoingHttpHeader) => {
   return String(header);
 };
 
-export function parseCookies(
+/**
+ * Parses a (coma-separated) list of Set-Cookie headers
```
Comment (sommeeeer) [inline packages/open-next/src/http/util.ts]: comma

File: packages/open-next/src/http/util.ts [hunk 2]
```diff
@@ -28,7 +28,13 @@ export const convertHeader = (header: http.OutgoingHttpHeader) => {
...
+ *
+ * @param cookies A coma-separated list of Set-Cookie headers or a list of Set-Cookie header
+ * @returns A list of Set-Cookie header
```
Comment (vicb) [inline packages/open-next/src/http/util.ts]:
```suggestion
 * Parses a (comma-separated) list of Set-Cookie headers
 *
 * @param cookies A comma-separated list of Set-Cookie headers or a list of Set-Cookie header
 * @returns A list of Set-Cookie header
```

File: packages/open-next/src/http/util.ts (hunk 1 above)
Comment (vicb) [reply packages/open-next/src/http/util.ts]: Thanks!

File: packages/open-next/src/http/util.ts [hunk 3]
```diff
@@ -28,7 +28,13 @@ export const convertHeader = (header: http.OutgoingHttpHeader) => {
   return String(header);
 };
 
-export function parseCookies(
+/**
+ * Parses a (comma-separated) list of Set-Cookie headers
+ *
+ * @param cookies A comma-separated list of Set-Cookie headers or a list of Set-Cookie header
```
Comment (vicb) [inline packages/open-next/src/http/util.ts]:
```suggestion
 * @param cookies A comma-separated list of Set-Cookie headers or a list of Set-Cookie headers
```

**Comment by sommeeeer** [comment]
LGTM

Thanks for doing this. We can consider adding an unit test for the edge converter at a later time.

**Comment by vicb** [comment]
> Thanks for doing this. We can consider adding an unit test for the edge converter at a later time.

I'll look into adding an e2e...
_(there were unit tests for the Node impl)_

File: packages/open-next/src/http/util.ts [hunk 4]
```diff
@@ -28,7 +28,13 @@ export const convertHeader = (header: http.OutgoingHttpHeader) => {
...
 };
 
-export function parseCookies(
+/**
+ * Parses a (comma-separated) list of Set-Cookie headers
+ *
+ * @param cookies A comma-separated list of Set-Cookie headers or a list of Set-Cookie headers
+ * @returns A list of Set-Cookie header
+ */
+export function parseSetCookieHeader(
```
Comment (khuezy) [inline packages/open-next/src/http/util.ts:37]: 👍 In hindsight, there should have been a comment on the "Expires-" regex, it can be cryptic if you don't have the context.

TL;DR: it splits each cookie on the "," but not when the comma belongs to the `Expires=Thu, 01 June ....` value.

Comment (vicb) [reply packages/open-next/src/http/util.ts:37]: I wanted to add that but forgot, thanks for the reminder :)

**Comment by vicb** [comment]
Thanks for the reviews!
